diff --git a/src/lib.rs b/src/lib.rs
index 97cb8d4..2de2726 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -698,10 +698,7 @@ impl<'h, 'b> Response<'h, 'b> {
                 bytes.slice();
                 self.reason = Some("");
             },
-            b'\n' => {
-                bytes.slice();
-                self.reason = Some("");
-            }
+            b'\n' => self.reason = Some(""),
             _ => return Err(Error::Status),
         }
 
